### PR TITLE
pbw-scss

### PR DESCRIPTION
**Pull request by mistermendez**

### CODE REVIEW

File: js/components/sharePanel.js [hunk 1]
```diff
@@ -88,143 +64,110 @@ var SharePanel = React.createClass({
...
-      this.setState({activeTab: panelToShow});
-    }
+    this.setState({activeTab: panelToShow});
   },
 
   render: function() {
-    var shareStyle = InlineStyle.shareScreenStyle;
-    var activeTabStyle = Utils.extend(shareStyle.tabStyle, shareStyle.activeTab);
-    var activeLastTabStyle = Utils.extend(shareStyle.lastTabStyle, shareStyle.activeTab);
+    var shareTab = ClassNames({
```
Comment (mistermendez) [inline js/components/sharePanel.js]: @lental: a cool utility for managing css class names in react

File: js/components/sharePanel.js [hunk 2]
```diff
@@ -88,143 +64,110 @@ var SharePanel = React.createClass({
...
+            </div>
+
+          </form>
         </div>
       );
     }
   },
 
   handleEmailClick: function(evt) {
-    if (evt.type == 'touchend' || !this.isMobile){
```
Comment (mistermendez) [inline js/components/sharePanel.js]: use <a> or <button> to prevent the need for this event check

File: js/components/sharePanel.js (hunk 1 above)
Comment (lental) [inline js/components/sharePanel.js]: :+1: :+1:

File: js/views/shareScreen.js [hunk 3]
```diff
@@ -30,34 +20,13 @@ var ShareScreen = React.createClass({
...
-  closeSharePanel: function(evt) {
-    if (evt.type == 'touchend' || !this.isMobile){
-      //since mobile would fire both click and touched events,
-      //we need to make sure only one actually does the work
-
-      this.props.controller.closeScreen();
-    }
-  },
-
-  highlight: function(evt) {
```
Comment (mistermendez) [inline js/views/shareScreen.js]: replaced with :hover in css

File: js/views/shareScreen.js [hunk 4]
```diff
@@ -30,34 +20,13 @@ var ShareScreen = React.createClass({
...
-  removeHighlight: function(evt) {
-    var opacity = "0.6";
-    Utils.removeHighlight(evt.target, opacity);
-  },
-
   render: function() {
     return (
-      <div className="ShareScreen" style={{height: "100%", width: "100%"}}>
-        <SharePanel {...this.props}/>
-        <div className="closeButton" onMouseOver={this.highlight} onMouseOut={this.removeHighlight}
```
Comment (mistermendez) [inline js/views/shareScreen.js]: moved to reusable component - /js/components/closeButton.js

File: sample.html [hunk 5]
```diff
@@ -4,7 +4,7 @@
   <script src="//player.ooyala.com/static/v4/latest/core.min.js"></script>
   <script src="//player.ooyala.com/static/v4/latest/discovery_api.min.js"></script>
   <script src="//player.ooyala.com/static/v4/latest/html5-skin/build/html5-skin.min.js"></script>
-  <link rel="stylesheet" href="//player.ooyala.com/static/v4/latest/html5-skin/assets/styles.css"/>
+  <link rel="stylesheet" href="//player.ooyala.com/static/v4/latest/html5-skin/build/html5-skin.css"/>
```
Comment (mistermendez) [inline sample.html]: new CSS file compiled from SCSS

File: js/components/sharePanel.js (hunk 1 above)
Comment (mistermendez) [inline js/components/sharePanel.js]: https://github.com/JedWatson/classnames

used to be bundled w/ react-tools

File: README.md [hunk 6]
```diff
@@ -85,6 +85,45 @@ To start a python server, cd into the repo directory and run:
...
 
+## Style
+We use [Sass](http://sass-lang.com/) (SCSS) for CSS preprocessor and [Bootstrap](http://getbootstrap.com/) for responsive, mobile design.
+
+Our [4-1 architecture pattern](http://sass-guidelin.es/#the-7-1-pattern) splits the Sass codebase over several files that are compiled into a single, minified stylesheet deployed to production.
+
+This approach maintains modular, decoupled style without impacting performance.
+
+```
+scss/
```
Comment (mistermendez) [inline README.md]: @gregorymostizky @lental @aldigun @chibzu @ksunika @rchalooyala 

Updated README explains Sass architecture pattern -

File: js/components/sharePanel.js [hunk 7]
```diff
@@ -15,70 +7,54 @@ var React = require('react'),
...
-      activeTab: this.tabs.SHARE,
+      activeTab: this.tabs.SHARE
     };
   },
 
   getActivePanel: function() {
-    var shareStyle = InlineStyle.shareScreenStyle;
     if (this.state.activeTab === this.tabs.SHARE) {
-      var twitterIconStyle = Utils.extend(shareStyle.socialIconStyle, shareStyle.twitterIconStyle);
-      var facebookIconStyle = Utils.extend(shareStyle.socialIconStyle, shareStyle.facebookIconStyle);
```
Comment (mistermendez) [inline js/components/sharePanel.js]: replace by Sass @extend function

File: js/components/sharePanel.js [hunk 8]
```diff
@@ -88,143 +64,110 @@ var SharePanel = React.createClass({
...
-
-      this.setState({activeTab: panelToShow});
-    }
+    this.setState({activeTab: panelToShow});
   },
 
   render: function() {
-    var shareStyle = InlineStyle.shareScreenStyle;
-    var activeTabStyle = Utils.extend(shareStyle.tabStyle, shareStyle.activeTab);
-    var activeLastTabStyle = Utils.extend(shareStyle.lastTabStyle, shareStyle.activeTab);
```
Comment (mistermendez) [inline js/components/sharePanel.js]: replace by Sass @extend function

File: scss/components/_share-screen.scss [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+      border-radius: 6px;
+    }
+
+    .twitter {
+      @extend .social-icon-style;
+      background-color: #64ccea;
+    }
+
+    .facebook {
+      @extend .social-icon-style;
```
Comment (mistermendez) [inline scss/components/_share-screen.scss]: replaces Utils.extend()

**Comment by mistermendez** [comment]
@aldigun conflicts resolved

**Comment by aldigun** [comment]
my concern is the amount of file added and the resulting of exponential file size increase. Are those files meet the bare minimum to make it work?

File size is a valid concern and will definitely going to be one of major factors in customer choosing this skin.

**Comment by mistermendez** [comment]
@aldigun reduced css to bare minimum

removed bootstrap styles for  - responsive grid, forms, normalization, resets, gradients, buttons

File: js/components/sharePanel.js [hunk 10]
```diff
@@ -1,13 +1,5 @@
-/********************************************************************
-  SHARING SCREEN
-*********************************************************************/
-/**
-* The screen used while the video is playing.
-*
-* @class PlayingScreen
-* @constructor
-*/
```
Comment (aldigun) [inline js/components/sharePanel.js]: We should not remove JSDOC comment.

File: gulpfile.js [hunk 11]
```diff
@@ -58,11 +59,11 @@ gulp.task('pretty', function () {
...
 
-// Build CSS
-gulp.task('buildCss', function() {
-  gulp.src(path.css)
-  //.pipe(concat('html5-skin.css'))
-  .pipe(gulp.dest('build'));
+// Build Sass
+gulp.task('sass', function () {
+  gulp.src(path.sass)
+    .pipe(sass({outputStyle: 'compressed'}).on('error', sass.logError))
```
Comment (aldigun) [inline gulpfile.js]: We need to have a 2 versions of min and un-minified of css files.
please read the guideline on naming convention

File: gulpfile.js [hunk 12]
```diff
@@ -10,59 +10,69 @@ var gulp = require('gulp'),
...
-    .pipe(source('html5-skin.min.js'))
+    .pipe(source('html5-skin.js'))
     .pipe(buffer())
-    .pipe(sourcemaps.init({loadMaps: true}))
-    // Add transformation tasks to the pipeline here.
-    .pipe(uglify())
     .on('error', gutil.log)
-    .pipe(sourcemaps.write('./'))
-    .pipe(gulp.dest('./build/'));
+    .pipe(gulp.dest('./build/js'));
```
Comment (aldigun) [inline gulpfile.js]: Do we need to create a new folder of js and css inside build folder?

Comment (mistermendez) [inline gulpfile.js]: it's getting crowded with 6 flies that all have the same 'html-5-skin' name

**Comment by mistermendez** [comment]
@aldigun paths updated based on new deployment guidelines

**Comment by mistermendez** [comment]
@aldigun @rchalooyala output destination for built files switched back to /build

File: scss/base/_type.scss [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+    font-style: normal;
+
+  }
+  /* roboto condensed regular */
+  @font-face {
+    font-family: 'Roboto Condensed';
+    font-style: normal;
+    font-weight: 400;
+    src: local('Roboto Condensed'), local('RobotoCondensed-Regular'),
+    url(/assets/fonts/RobotoCondensed-Regular.woff2) format('woff2'),
```
Comment (aldigun) [inline scss/base/_type.scss]: Is this the correct link? you are hard-coding the URL reference

**Comment by mistermendez** [comment]
@aldigun conflicts resolved

**Comment by aldigun** [comment]
All set!

**Pull request merged**